src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -2096,6 +2096,13 @@ void Unit::AttackerStateUpdate(Unit* victim, WeaponAttackType attType, bool extr
     if (HasUnitState(UNIT_STATE_CANNOT_AUTOATTACK) && !extra)
         return;
 
+    if (HasUnitState(UNIT_STATE_CASTING))
```

Comment:
we should consider moving this block into a separate method and then just call it here and in `UnitAI::DoMeleeAttackIfReady()` as its literally the same code :)

Reply:
autoattack code unification can wait